Public/Src/Cache/ContentStore/App/DistributedService.cs

```diff
@@ -118,9 +118,40 @@ public void OnTeardownCompleted()
             {
             }
 
-            public Task<Dictionary<string, string>> RetrieveKeyVaultSecretsAsync(List<string> secrets, CancellationToken token)
+            public Task<Dictionary<string, Credentials>> RetrieveKeyVaultSecretsAsync(List<RetrieveSecretsRequest> requests, CancellationToken token)
             {
-                return Task.FromResult(secrets.ToDictionary(s => GetSecretStoreValue(s)));
+                var result = new Dictionary<string, Credentials>();
+
+                foreach (var request in requests)
+                {
+                    Credentials credentials = null;
+
+                    var secret = GetSecretStoreValue(request.Name);
+                    if (string.IsNullOrEmpty(secret))
+                    {
+                        // Environment variables are null by default. Skip if that's the case.
+                        continue;
+                    }
+
+                    switch (request.Kind)
+                    {
+                        case CredentialsKind.AzureBlobPlainText:
+                            credentials = new AzureBlobStorageCredentials(secret);
+                            break;
+                        case CredentialsKind.EventHubPlainText:
+                            credentials = new EventHubCredentials(secret);
+                            break;
+                        case CredentialsKind.RedisPlainText:
+                            credentials = new RedisCredentials(secret);
+                            break;
+                        case CredentialsKind.AzureBlobSASToken:

```

Comment:
Could you explain why this is not supported?

I thought we can easily implement this already here... What prevents us from doing that?